### ISSUE TITLE
No global element for root was found

### ISSUE THREAD

**Issue by hdehaan01** [pepri/wizdler#12]
I get the error in Wizdler when invoking a function:
No global element for root was found: heartbeatRequest (http://www.amorgroup.com/chroma/webservices/1.0/flightlegdataservice)

I have selected a wsdl. This wsdl contains 4 functions.

**Comment by pepri** [comment]
Please provide the WSDL so I can look into it.

**Comment by hdehaan01** [comment]
I tried that, but it only accepts images. How should I do that?

**Comment by pepri** [comment]
You can either paste the WSDL content here, or you can send it to my e-mail: peto.prikryl@gmail.com

**Comment by hdehaan01** [comment]
Peter Prikryl schreef op 17.03.2015 15:07:

> You can either paste the WSDL content here, or you can send it to my
> e-mail: peto.prikryl@gmail.com
> 
> ## 
> 
> Reply to this email directly or view it on GitHub [1].
> 
>   *
> 
> ## Links:
> 
> [1] https://github.com/pepri/wizdler/issues/12#issuecomment-82371464

I have packed the directory in a rar. I used the 
FlightLegDataService.wsdl

Thank you in advance.

Hans de Haan

**Comment by pepri** [comment]
I have not received any RAR file yet.

**Comment by hdehaan01** [comment]
It should come with a mail

**Comment by hdehaan01** [comment]
Did you receive the mail?

**Comment by pepri** [comment]
No, I have not. I will let you know if I do.

**Comment by hdehaan01** [comment]
Peter Prikryl schreef op 17.03.2015 15:33:

**Comment by pepri** [comment]
Ok, I received the WSDL, but it imports another files in xs:import elements. I need all the referenced files to analyze the issue.

**Comment by hdehaan01** [comment]
I did sent you twice a rar file, but it looks like it does not get through to your server. The rar file contains all xsd definitions.

Any idea how we can get it to you?

**Comment by pepri** [comment]
The WSDL references XSD "xsd/flightlegdata/FlightLegDataService.xsd". Be sure it is available. Without it, the message "No global element for root was found" will be shown. You can check that Wizdler tried to download that file in the Web inspector on the Network tab after refreshing the editor with F5. If you are sure it is available (there is no error in the console), then I will have to look further into it.

Not sure, what the problem with the e-mail is. Maybe your SMTP server does not allow attaching RAR files. Try to rename it to TXT without packing into RAR or ZIP. Or you can just paste the content into some text file or directly into the e-mail body.

**Comment by hdehaan01** [comment]
Hello Peter,

The xsd is available, but in a sub-directory. That is defined in the wsdl:   

 <types>

```
    <xs:schema xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/">

        <xs:import schemaLocation="xsd/flightlegdata/FlightLegDataService.xsd"

                   namespace="http://www.amorgroup.com/chroma/webservices/1.0/flightlegdataservice" />

    </xs:schema>

</types>
```

Although I tried to send the xsd definitions, I got aware that these files are part of a non-disclosure with the party that delivered these files.

What I will do is simplify the definitions to a single xsd and perform the test over.

I will send you the result later.

Is that ok to you?

Thanks in advance.

Best regards,

Hans de Haan

**Comment by pepri** [comment]
I am ok with any complete test case that I can reproduce. The XSD can import/include other XSDs so the type `heartbeatRequest` can be defined in yet another XSD schema.

What we basically have to do is to find out, why the type cannot be found - most probably the XSD with that file was not loaded. You should see the requests for XSDs in the web inspector on the network tab. Wizdler does not support XSD include (I did not need the XSD include yet), so that may be the source of this issue.

**Comment by kmila** [comment]
This issue also occurs with the WSDL mentioned on [this comment](http://stackoverflow.com/questions/3729425/looking-for-a-tool-to-flatten-wsdl-and-its-xsd#comment48780688_7111740) from stackoverflow.

**Comment by pepri** [comment]
Fixed in 1.28.
